src/export/dk.ml

```diff
@@ -0,0 +1,183 @@
+(** Export a Lambdapi signature to Dedukti. *)
+
+open Lplib open Base open Extra
+open Timed
+open Common
+open Core open Term
+
+let string = Format.pp_print_string
+
+(** Translation of identifiers. Lambdapi identifiers that are Dedukti keywords
+   or invalid Dedukti identifiers are escaped, a feature offered by
+   Dedukti. *)
+
+let keyword_table = Hashtbl.create 59
+
+let is_keyword : string -> bool = Hashtbl.mem keyword_table
+
+let _ = let open Parsing.DkTokens in
+  let loc = Lexing.dummy_pos, Lexing.dummy_pos in
+  List.iter (fun (s, t) -> Hashtbl.add keyword_table s t)
+    [ ".", DOT
+    ; ",", COMMA
+    ; ":", COLON
+    ; "==", EQUAL
+    ; "[", LEFTSQU
+    ; "]", RIGHTSQU
+    ; "{", LEFTBRA
+    ; "}", RIGHTBRA
+    ; "(", LEFTPAR
+    ; ")", RIGHTPAR
+    ; "-->", LONGARROW
+    ; "->", ARROW
+    ; "=>", FATARROW
+    ; ":=", DEF
+    ; "Type", TYPE loc
+    ; "def", KW_DEF loc
+    ; "defac", KW_DEFAC loc
+    ; "defacu", KW_DEFACU loc
+    ; "injective", KW_INJ loc
+    ; "thm", KW_THM loc
+    ; "private", KW_PRV loc
+    ; "#NAME", NAME (loc, "")
+    ; "#REQUIRE", REQUIRE (loc, "")
+    ; "#EVAL", EVAL loc
+    ; "#INFER", INFER loc
+    ; "#CHECK", CHECK loc
+    ; "#CHECKNOT", CHECKNOT loc
+    ; "#ASSERT", ASSERT loc
+    ; "#ASSERTNOT", ASSERTNOT loc
+    ; "#PRINT", PRINT loc
+    ; "#GDT", GDT loc]
+
+let is_ident : string -> bool = fun s ->
+  Parsing.DkLexer.is_ident (Lexing.from_string s)
+
+let escape : string pp = fun ppf s -> out ppf "{|%s|}" s
+
+let ident : string pp = fun ppf s ->
```

Comment:
What happens when the symbol uses utf8? Does Dedukti handles utf8 in symbols?

Reply:
The symbol is escaped.